### ISSUE TITLE
Browser unsupported for push notifications on mobile.twitter.com

### ISSUE THREAD

**Issue by ronakg** [jiahaog/nativefier#652]
### Description

I created an app for mobile.twitter.com. When I visit Settings -> Push Notifications -> Enable, it says that the browser is not supported. 

### Steps to reproduce issue

$ nativefier --name Twitter --counter --max-width 600 -f --single-instance mobile.twitter.com

1. login to the app using your Twitter credentials
2. Click profile icon
3. Go to Settings and Privacy -> Notifications -> Push Notifications
4. Click Turn On.

### Details

- Are you nativefying a *public* website? : `Yes`
- Feature request? Have you looked at `nativefier --help` to see if an existing option could fit your needs?: Yes
- Full nativefier command used to build your app: `nativefier --name Twitter --counter --max-width 600 -f --single-instance mobile.twitter.com`
- Version of Nativefier (run `nativefier --version`): `v7.6.4`
- Version of node.js (run `node --version`): `v5.12.0`
- OS: `MacOS 10.13.5`
- Error message / stack trace (if any): `None`

**Comment by ronjouch** [comment]
@ronakg thanks for the feedback. Can you try checking what the reported User Agent is (pop up the DevTools, then evaluate `navigator.userAgent`) and spoof it to a regular browser one via the `--user-agent` flag?

**Comment by ronakg** [comment]
`navigator.userAgent` is set to `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36`.

I've already tried setting `--user-agent 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36'`, which didn't help.

**Comment by ronakg** [comment]
On the regular browser window, I get a pop-up asking if the website should be allowed to send notifications. That window is not showing up for the nativefier app.

**Comment by alexey13** [comment]
The same as @ronakg has

**Comment by BananaManCJ** [comment]
Any update on this? I still can't get it to work.

**Comment by ronjouch** [comment]
> Any update on this? I still can't get it to work.

@BananaManCJ oh hey, thanks, this issue is a dup of https://github.com/jiahaog/nativefier/issues/88 . Notifications were broken but fixed by @exander77 in 1d3bed5f09961e878c4c40085bd69aa2159d3a4c, but this is unreleased yet, as I'm waiting on one thing (see below). If you really cannot wait, [install a dev. build](https://github.com/jiahaog/nativefier/blob/master/docs/development.md).

@exander77, you said you'd add a flag to enable/disable notifications, and keep the current default to disabled notifications, as doing the contrary would be breaking current behavior established since 2016. Are you still progressing on this?

**Comment by BananaManCJ** [comment]
Notifications are still not working on Twitter

**Comment by ronjouch** [comment]
Please read my above message. Thanks.

**Comment by BananaManCJ** [comment]
I think the problem is with it authorizing notifications, notifications do work on http://www.bennish.net/web-notifications.html

**Comment by exander77** [comment]
@BananaManCJ Can you verify this page?
https://www.bennish.net/web-notifications.html
If it works ok, then I will investigate what is going on it Twitter.
If it does not work, then there is something strange on your end, like not actually using the dev build.

**Comment by exander77** [comment]
Oh, you have already tried bennish test side, then it is something strange in Twitter, will check.

**Comment by BananaManCJ** [comment]
I'm using the test build and the debugger is giving me no useful information

**Comment by BananaManCJ** [comment]
> Please read my above message. Thanks.

Oops, forgot to mention that I was using the test build and not the release

**Comment by exander77** [comment]
@BananaManCJ How can I easily create a notification on Twitter?

**Comment by BananaManCJ** [comment]
> @BananaManCJ How can I easily create a notification on Twitter?

What's your Twitter, I can try sending you a dm

**Comment by exander77** [comment]
Hmm, seems like "push notifications" are not standard "notifications", it is something different for mobile devices. Can you please elaborate for me how it actually differs from normal notifications?

**Comment by BananaManCJ** [comment]
I'm not entirely sure how it works, but you might be able to find a javascript file

**Comment by BananaManCJ** [comment]
https://abs.twimg.com/responsive-web/web/bundle.Notifications.0e522c64.js
https://abs.twimg.com/responsive-web/web/bundle.NotificationDetail.24bb4ac4.js
https://abs.twimg.com/responsive-web/web/loader.PushNotificationsPrompt.660083f4.js

Got these from https://mobile.twitter.com/sw.js

**Comment by BananaManCJ** [comment]
I am also seeing that Twitter uses service workers for it's notifications

**Comment by exander77** [comment]
This will need a more in-depth analysis, Push Notifications seems to be initially supported on mobile devices, but now are also available in Desktop browsers. It is a completely different API from normal Notifications.

There seems to be recent work on this module: https://github.com/MatthieuLemoine/electron-push-receiver
For Electron, but I am not sure what is a state of it.

**Comment by BananaManCJ** [comment]
Yeah, I've seen that before, but I wasn't sure how to implement it.

**Comment by BananaManCJ** [comment]
@exander77 You still planning on making a flag to disable and enable notifications?

**Comment by exander77** [comment]
@BananaManCJ Yes, should go back to it. I was actually looking into those pushdown notifications, but it seems like quite complicated things.

**Comment by BananaManCJ** [comment]
I don't know if you saw this but there is a working example [here](https://github.com/CydeSwype/electron-fcm-demo)

**Comment by BananaManCJ** [comment]
Does Nativefier support service workers?

**Comment by BananaManCJ** [comment]
@exander77 it's been over a month since a commit, are you still working on adding that cli flag?

**Comment by CoreyVidal** [comment]
Has this been resolved? I'm trying Twitter.com and it's not allowing me to enable notifications.

```
nativefier --name Twitter --single-instance --tray start-in-tray --show-menu-bar --verbose "https://twitter.com" "C:\Program Files\Nativefier"
```

**Comment by BananaManCJ** [comment]
@CoreyVidal Twitter uses GCM for push notifications, which is not implemented in Electron at the moment.
